src/core/providers/ogr/qgsogrdataitems.cpp

```diff
@@ -627,8 +627,8 @@ QgsDataItem *QgsOgrDataItemProvider::createDataItem( const QString &pathIn, QgsD
     const auto constWildcards = QgsOgrProviderUtils::wildcards();
     for ( const QString &wildcard : constWildcards )
     {
-      QRegExp rx( wildcard, Qt::CaseInsensitive, QRegExp::Wildcard );
-      if ( rx.exactMatch( info.fileName() ) )
+      QRegularExpression rx( QRegularExpression::wildcardToRegularExpression( wildcard ), QRegularExpression::CaseInsensitiveOption );
```

Comment:
ifdef -- also needs to be an anchored pattern

Reply:
```suggestion
      QRegularExpression rx( QRegularExpression::anchoredPattern( wildcard ), QRegularExpression::CaseInsensitiveOption );
```

is this correct?

Reply:
Not sure -- you'd need to test the results :)